modules/trusted-firmware-m/src/zephyr_tfm_psa_test.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+/**
+ * \brief This symbol is the entry point provided by the PSA API compliance
+ *        test libraries
+ */
+extern void val_entry(void);
+
+
+void psa_test(void)
+{
+#if defined(CONFIG_TFM_PSA_TEST_NONE)
+	printk("No PSA test suite set. Use Kconfig to enable a test suite.\n");
```

Comment:
Should there be a build error here? Just asking :)

Reply:
This file is always included when TFM is included. In this way, the error is only printed if the function is called.

Reply:
Well. I assume that would depend on whether we do:
```
zephyr_sources(..../src/zephyr_tfm_psa_test.c)
```
as currently suggested in the PR, or we decide to do:
```
zephyr_sources_ifndef(CONFIG_TFM_PSA_TEST_NONE ..../src/zephyr_tfm_psa_test.c)
```

so what is the reason for always including the file in the build ?
Is TFM always linking to this function even if it is never called.

Reply:
Fixed

Reply:
but code is still here ?
In #34380 you don't include the file when `CONFIG_TFM_PSA_TEST_NONE=n` so I guess this code could be removed.